src/peft/tuners/lora/config.py

```diff
@@ -346,6 +347,26 @@ def __post_init__(self):
             if self.loftq_config is None:
                 raise ValueError("`loftq_config` must be specified when `init_lora_weights` is 'loftq'.")
 
+        # Using post training conversion of modified base weights to restore their initial values (PiSSA, OLoRA) cannot
+        # be correctly done when using rslora + rank_pattern/alpha_pattern. We can't really know if the user intends
+        # this when they'll eventually call save_pretrained (i.e. if they'll pass
+        # path_initial_model_for_weight_conversionl). Therefore, we only warn but don't raise an error here.
+        if (
+            self.use_rslora
+            and (self.rank_pattern or self.alpha_pattern)
+            and (
+                (isinstance(self.init_lora_weights, str) and (self.init_lora_weights.startswith("pissa")))
+                or (self.init_lora_weights == "olora")
+            )
+        ):
+            msg = (
+                "Using Rank-Stabilized LoRA with rank_pattern/alpha_pattern and post-training conversion of modified "
+                "base weights (PiSSA, OLoRA) means that you won't be able to pass "
+                "`path_initial_model_for_weight_conversion` to `save_pretrained` to restore the initial values of the "
+                "base weights; if you intend to do this, please ensure not to use rslora or rank_pattern/alpha_pattern."
+            )
+            warnings.warn(msg)
```

Comment:
Maybe we can document this limitation in the API spec itself, as well? 

I would also remember to include this in the release notes so that users are firmly aware of this.

Reply:
I'll add a mention to the docstring of `save_pretrained`, as well as to the README files of the two methods.

I added an entry to the PEFT release notes to mention this change there as well. As OLoRA is as yet unreleased and PiSSA was released in the last version, and also given that this is an edge case, I think very few users, if any, will be affected.